### DISCUSSION TITLE
Z-Offset Issues

### DISCUSSION THREAD

**Discussion by shawnb2** [Jyers/Marlin#93] [Q&A] [answered]
Please excuse my major noobness as I'm not very savvy with GitHub or Reddit and how it all works. I'm running this firmware on my V2 and overall it's awesome but I do have an issue and want to see if you could help. I'm running the version with BLTouch and fast probing but when I went to set z-offset it wouldn't adjust in live time. I would have to get the nozzle close at first and then dial it in a couple hundredths at a time but it wouldn't adjust after I clicked the wheel. Would have to back out of the screen and then it would adjust and would go back to recheck for proper height. Rinse and repeat. I unfortunately didn't fully figure this out until I sacrificed my new PEI Sheet. I have it set now and it's works great but wanted to see if this is a known issue that could be fixed or if I'm the lucky one. Either way, if there's a way to fix it I would greatly appreciate it. Thanks.

**Comment by Jyers** [comment, marked as answer]
Just a sanity check here... Did you enable "Live Adjust"?

**Reply by shawnb2** [reply]
I did at some point but may have not been using it as was expected. Someone else in one of the FB groups mentioned that as well and it very well could be my mistake because I wasn't familiar with this setup. My other printers which includes a Prusa Mini and an E3 Pro with the 4.2.7 board running TH3D firmware have only ever had to click on the value to enable it and adjust on the fly. Never had to enable with one menu and then enable again actually clicking on the value number so if that's truly all it is then it's just a user error on my part. Thanks for checking my sanity. Haha. Needs to be done once in awhile.

**Reply by Jyers** [reply]
The reason for the separate live adjust setting is so that if you just wanted to adjust the z offset a bit and didn't want to have to re-home the z axis, you can just adjust it without live adjust enabled.

**Reply by shawnb2** [reply]
Ok thank you. It was clearly just my not being familiar with how it was setup and if there is a guide somewhere I'd be happy to look it over but had just gone in blind because someone had posted about it. Thanks for responding and clearing things up for me. I really appreciate it. It's definitely hands down the best firmware available for the V2 hands down.

**Reply by Jyers** [reply]
I believe there is someone who should be making a video guide on this firmware, so hopefully soon people won't have to go in blind :)

**Reply by npike** [reply]
When Live Adjust is enabled in the z-offset menu, should the nozzle be moving when I change the offset?  Mine seems to stay parked unless I re-home the z-offset after each change.

**Reply by Jyers** [reply]
Yes, it should be moving. If you continue to face this issue, feel free to create a bug report in the issue section of github.

**Reply by Schnabulation** [reply]
Is it needed to habe `BABYSTEP_ALWAYS_AVAILABLE` defined for the live-adjust to be working?

**Reply by Jyers** [reply]
As far as I know BABYSTEP_ALWAYS_AVAILABLE should not have any impact on the z offset menu. I have always used it without this enabled and it has never been an issue.
